macOS/DuckDuckGo/Permissions/Model/PermissionModel.swift

```diff
@@ -130,7 +142,22 @@ final class PermissionModel {
                     permissions.geolocation
                         .systemAuthorizationDenied(systemWide: !geolocationService.locationServicesEnabled())
                 } else {
-                    permissions.geolocation.update(with: webView.geolocationState)
+                    let currentState = webView.geolocationState
+
+                    // Track if geolocation was ever activated
+                    if currentState == .active {
+                        geolocationWasActivated = true
+                    }
+
+                    // With new permission view, keep geolocation as active once it's been used
+                    if featureFlagger.isFeatureOn(.newPermissionView),
+                       geolocationWasActivated,
+                       currentState == .none,
+                       permissions.geolocation == .active || permissions.geolocation == .inactive {
+                        permissions.geolocation = .active
```

Comment:
Geolocation was active just for a short period of time. I think the active state should represent that website read the value. Happy to adjust if you have a better solution to this in mind